controller/zuriInternController.js

```diff
@@ -1,16 +1,45 @@
 const ZuriIntern = require('../models/ZuriInternModel');
 const {
-  responseHandler
+    responseHandler
 } = require('../utils/responseHandler');
 
 module.exports = {
-  getAllInterns: async (req, res) => {
-    try {
-      const zuriInterns = await ZuriIntern.find();
-      return responseHandler(res, 'Success', 200, true, zuriInterns);
-    } catch (err) {
-      return responseHandler(res, 'Error', 500, false, err);
+    getAllInterns: async (req, res) => {
+        let searchValue;
+        if (req.query.firstName) {
+            searchValue = {
+                firstName: req.query.firstName
+            }
+        } else {
+            searchValue = {}
+        }
+        try {
+            const zuriInterns = await ZuriIntern.find(searchValue);
+            return responseHandler(res, 'Success', 200, true, zuriInterns);
+
+        } catch (err) {
+            return responseHandler(res, 'An Error occured', 500, false, err);
+        }
+    },
+
+    filterInterns: async (req, res) => {
```

Comment:
Please add this functionality to the same GET endpoint,you can use query params
/interns/?gender=male should return only male interns etc